lib/process.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
```

Comment:
Attempting to do `make` in `lib/read_only/dl_esm_inf` fails because this script is not executable. It also failes if dl_esm_inf has not already been compiled. Unless there's a good reason not to, please could you extend the Makefile to build the library too if necessary.

Reply:
OK, missing 'x' for process.py ... thanks to my windows file system (fwiw, I am going to switch to a native linux FS on my new VM, which has more disk space.) I'll add the dl_esm_inf compilation.

Your final compilation problem is then a left-over from missing execute permissions of process.py: it created an empty file, which was then compiled, creating ... nothing :) If you remove read_only_base.f90 (or do a ``make clean``) it should work. But I'll start working on the other bits and pieces tomorrow.

Reply:
Aha! I'd guessed it was your windows file system but didn't think to do a `make clean`. I've done that and now have built the dl_esm_inf read-only wrapper successfully. While I'm here, is there an example that demonstrates this?

Reply:
examples/gocean/eg7
I am having now some compilation problems there ... the created alg.f90 file contains >132 characters, so we either need to add ``-l output`` to psyclone, or the flag to gfortran to accept lines >132 characters (-ffree-line-length-none I believe).

Reply:
I'd go with the PSyclone solution if possible - try and minimise compiler-specific dependencies.

Reply:
Done.